src/stream.rs

```diff
@@ -253,36 +251,60 @@ impl UnitState {
     async fn update(&mut self, ctx: &Context) -> Result<bool, Status> {
         let mut changed = false;
 
+        #[derive(serde::Serialize)]
+        struct GetUpdateRequest {
+            name: String,
+        }
+
+        #[derive(Debug, serde::Deserialize)]
+        struct Velocity {
+            x: f64,
+            y: f64,
+            z: f64,
+        }
+
+        #[derive(Debug, serde::Deserialize)]
+        struct Update {
+            position: Position,
+            velocity: Velocity,
+        }
+
+        let res: Update = ctx
+            .rpc
+            .request(
+                "getUnitUpdate",
+                Request::new(GetUpdateRequest {
+                    name: self.unit.name.clone(),
+                }),
+            )
+            .await?;
+
         // update position
-        let position = UnitService::get_position(
-            &ctx.rpc,
-            Request::new(unit::v0::GetPositionRequest {
-                name: self.unit.name.clone(),
-            }),
-        )
-        .await?
-        .into_inner()
-        .position;
-        if let (Some(a), Some(b)) = (position.as_ref(), self.unit.position.as_ref()) {
-            if !eq(a, b) {
+        if let Some(position) = &self.unit.position {
+            if !eq(&res.position, position) {
                 changed = true;
-                self.unit.position = position;
+                self.unit.position = Some(res.position);
             }
+        } else {
+            changed = true;
+            self.unit.position = Some(res.position);
         }
 
-        // update player name
-        let player_name = UnitService::get_player_name(
-            &ctx.rpc,
-            Request::new(unit::v0::GetPlayerNameRequest {
-                name: self.unit.name.clone(),
-            }),
-        )
-        .await?
-        .into_inner()
-        .player_name;
-        if player_name != self.unit.player_name {
```

Comment:
Not sure what my thoughts were, but I don't think the player name can change during the life-time of a unit, or can it? (thus removed and considered static)

Reply:
I think you are correct here.